api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
Getting agreement on something will be progress; we can refactor later
On Fri, Mar 27, 2015 at 9:09 PM Derek Parker <notifications@github.com>
wrote:

> In api/types.go
> <https://github.com/derekparker/delve/pull/88#discussion_r27340830>:
>
> > @@ -0,0 +1,47 @@
> > +package api
> > +
> > +type Context struct {
> > +	State DebuggerState `json:"state"`
>
> Right, that's more what I had in mind. I'm certainly up for another
> pattern if you've seen it work before, I simply haven't had the "ah-ha"
> moment as you've mentioned. And GH issues aren't the best sometimes for
> discussion design patterns, hah.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/derekparker/delve/pull/88/files#r27340830>.
>